pkg/network/protocols/http/monitor.go

```diff
@@ -234,8 +236,10 @@ func (m *Monitor) Start() error {
 		return err
 	}
 
-	// Need to explicitly save the error in `err` so the defer function could save the startup error.
-	return m.processMonitor.Initialize()
+	if m.httpTLSEnabled {
+		err = m.processMonitor.Initialize()
+	}
+	return err
```

Comment:
```suggestion
	if m.httpTLSEnabled {
		return m.processMonitor.Initialize()
	}
	return nil
```

Reply:
err need to save as a variable, as defer func() check the err variable